src/plugins/discover/public/plugin.ts

```diff
@@ -277,6 +277,14 @@ export class DiscoverPlugin
       return `#${path}`;
     });
     plugins.urlForwarding.forwardApp('context', 'discover', (path) => {
+      const urlParts = path.split('/');
+      // take care of urls containing legacy url, those split in the following way
+      // ["", "context", indexPatternId, _type, id + params]
+      if (urlParts[4]) {
+        // remove _type part
+        const newPath = [...urlParts.slice(0, 3), ...urlParts.slice(4)].join('/');
+        return `#${newPath}`;
+      }
```

Comment:
@flash1293 adapted the logic .. much better without a nasty hack 👍 
one thing is to mention, 
what would break is a URL like , don't think there's a way to omit this
`/content/indexpattern/idWithSlash`
what would work is 
`/content/indexpattern/_doc/idWithMultipleSlash`